src/modules/noit_lua_noit_binding.c

```diff
@@ -126,12 +142,210 @@ lua_noit_checks_unsubscribe(lua_State *L) {
+          return 1;
+        case METRIC_UINT32:
+        case METRIC_UINT64:
+          lua_pushinteger(L, metric_id->value.v_int64);
+          return 1;
+        case METRIC_DOUBLE:
+          lua_pushnumber(L, metric_id->value.v_double);
+          return 1;
+        case METRIC_STRING:
+          lua_pushstring(L, metric_id->value.v_string);
+          return 1;
+      }
+      break;
+    default:
+      break;
+  }
+  luaL_error(L, "metric_value_t no such element: %s", k);
+  return 0;
+}
+
+static void
+noit_lua_setup_metric_value(lua_State *L,
+    metric_value_t *value) {
+  metric_value_t **addr;
+  addr = (metric_value_t **)lua_newuserdata(L, sizeof(value));
+  *addr = value;
+  if(luaL_newmetatable(L, "metric_value_t") == 1) {
+    lua_pushcclosure(L, noit_metric_value_index_func, 0);
+    lua_setfield(L, -2, "__index");
+  }
+  lua_setmetatable(L, -2);
+}
+
+static int noit_metric_id_index_func(lua_State *L) {
+  int n;
+  const char *k;
+  metric_id_t **udata, *metric_id;
+  n = lua_gettop(L); /* number of arguments */
+  assert(n == 2);
+  if(!luaL_checkudata(L, 1, "metric_id_t")) {
+    luaL_error(L, "metatable error, arg1 not a metric_id_t!");
+  }
+  udata = lua_touserdata(L, 1);
+  metric_id = *udata;
+  if(!lua_isstring(L, 2)) {
+    luaL_error(L, "metatable error, arg2 not a string!");
+  }
+
+  k = lua_tostring(L, 2);
+  switch (*k) {
+    case 'i':
+      if(!strcmp(k, "id")) {
+        char uuid_str[UUID_STR_LEN + 1];
+        uuid_unparse_lower(metric_id->id, uuid_str);
+        lua_pushstring(L, uuid_str);
+      } else if(!strcmp(k, "id_ud")) {
+        lua_pushlightuserdata(L, (void*)&metric_id->id);
+      } else {
+        return 0;
+      }
+      return 1;
+    case 'n':
+      if(!strcmp(k, "name")) {
+        lua_pushstring(L, metric_id->name);
+      } else if(!strcmp(k, "name_len")) {
+        lua_pushinteger(L, metric_id->name_len);
+      } else {
+        break;
+      }
+      return 1;
+    default:
+      break;
+  }
+  luaL_error(L, "metric_id_t no such element: %s", k);
+  return 0;
+}
+
+static void
+noit_lua_setup_metric_id(lua_State *L,
+    metric_id_t *msg) {
+  metric_id_t **addr;
+  addr = (metric_id_t **)lua_newuserdata(L, sizeof(msg));
+  *addr = msg;
+  if(luaL_newmetatable(L, "metric_id_t") == 1) {
+    lua_pushcclosure(L, noit_metric_id_index_func, 0);
+    lua_setfield(L, -2, "__index");
+  }
+  lua_setmetatable(L, -2);
+}
+
+static int
+noit_message_index_func(lua_State *L) {
+  int n;
+  const char *k;
+  metric_message_t **udata, *msg;
+  n = lua_gettop(L);    /* number of arguments */
+  assert(n == 2);
+  if(!luaL_checkudata(L, 1, "metric_message_t")) {
+    luaL_error(L, "metatable error, arg1 not a metric_message_t!");
+  }
+  udata = lua_touserdata(L, 1);
+  msg = *udata;
+  if(!lua_isstring(L, 2)) {
+    luaL_error(L, "metatable error, arg2 not a string!");
+  }
+
+  k = lua_tostring(L, 2);
+  switch(*k) {
+    case 'i':
+      noit_lua_setup_metric_id(L, &msg->id);
```

Comment:
this will allow sloppy usage like metric.id:isndifunsdf

you should

```
if(!strcmp(k, "id")) {
  noit_lua_setup_metric_id(L, &msg->id);
  return 1;
}
break;
```

And on all the subsequent ones as well.